### ISSUE TITLE
Feature request: Save views with templates

### ISSUE THREAD

**Issue by nategreen** [CodePen/CodePen-Bug-Reporting-Only#73]
Hey, guys! I love the new JS console and I am a big fan of the ability to start pens from a template. I think that feature would be enhanced if the view settings were saved with the template. That way, for instance, I could open a "theming" template and get only the CSS and JS panes, because I don't need to write any markup; or, I could open a "JS and console" template and *just* get the JS pane and the console.

What do you guys think?

**Comment by mariemosley** [comment]
I've made a note of your interest :) I'm closing this issue since this is not a bug. Thanks for using CodePen!

**Comment by nategreen** [comment]
Sorry about that! I was looking for the "leave feedback" thing on the site (I thought I'd used something like that before) but it looked like sending email was for serious matters only, so I came here.

**Comment by chriscoyier** [comment]
I kinda like the idea, for the record. But it's tricky, believe it or not. It's slightly at odd with user defaults. Like if you set your template to have certain view things how you want them, and someone else uses that template, who has different default user settings, which ones win? That's not unsolvable, but it bears a lot more thinking out than we've done. There are also so many categories of view-related things. Like which editor layout is just a temporal thing that remembers your last setting and uses that if it can. Tabs/Spaces is a per-pen thing and probably does make sense to follow the template (but currently doesn't). Other view preferences (e.g. line numbers) are user-settings which likely can't be overridden at a template level. Just thinking out loud!

**Comment by nategreen** [comment]
Yeah, that is a lot to consider. I didn't realize multiple people could use the same templates—I thought the templates I saved were just for me. Might be tomato-tomahto but I understood it this way: different users can mark *the same pen* as a template but the template is just for them. So, thinking of it that way, I would expect any of the settings that are in the pen settings modal to be saved per-pen and to change my template if they are altered. View settings that I adjust outside of that—layout, panel widths/heights, etc.—could be saved per-user-per-template…I would find it jarring to open another person's pen and for it to have their layout, for instance, but when I set up a template it's so I can shortcut all the little setup-y things I do before I start writing code/markup/whatever, so I personally wouldn't mind if it didn't remember the layout I had open a minute ago.

Anyway, that's my $0.02 but I probably don't speak for everyone. Thanks for the reply!
